src/CarbonAware.DataSources/CarbonAware.DataSources.ElectricityMaps/test/Client/ElectricityMapClientTests.cs

```diff
@@ -0,0 +1,457 @@
+        Assert.That(new DateTimeOffset(2099, 1, 1, 0, 0, 0, TimeSpan.Zero), Is.EqualTo(dataPoint?.CreatedAt));
+        Assert.That(dataPoint?.CarbonIntensity.ToString(), Is.EqualTo("999"));
+        Assert.That(dataPoint?.EmissionFactorType, Is.EqualTo(EmissionsFactor.Lifecycle));
+        Assert.IsFalse(dataPoint?.IsEstimated);
+        Assert.IsEmpty(dataPoint?.EstimationMethod);
+        Assert.That(new DateTimeOffset(2099, 1, 1, 0, 0, 0, TimeSpan.Zero), Is.EqualTo(dataPoint?.CreatedAt));
+    }
+
+    [Test]
+    public async Task GetHistoryAsync_FreeTrialToken_DeserializesExpectedResponse()
+    {
+        // Arrange
+        this.CreateHttpClient(m =>
+        {
+            if (m.RequestUri!.ToString().Contains(BaseUrls.TrialBaseUrl + Paths.History))
+            {
+                Assert.That(HttpMethod.Get, Is.EqualTo(m.Method));
+                var response = this.MockElectricityMapsResponse(m, new StringContent(TestData.GetHistoryCarbonIntensityDataJsonString()));
+                return Task.FromResult(response);
+            }
+            return Task.FromResult(this.MockElectricityMapsResponse(m, new StringContent("null")));
+        });
+
+        var freeTrialOptions = new Mock<IOptionsMonitor<ElectricityMapsClientConfiguration>>();
+        freeTrialOptions.Setup(o => o.CurrentValue).Returns(() => new ElectricityMapsClientConfiguration() { BaseUrl = BaseUrls.TrialBaseUrl });
+        var client = new ElectricityMapsClient(this.HttpClientFactory, freeTrialOptions.Object, this.Log.Object);
+
+        // Act
+        var data = await client.GetHistoryCarbonIntensityDataAsync(TestLatitude, TestLongitude);
+
+        // Assert
+        Assert.IsNotNull(data);
+        Assert.That(TestZone, Is.EqualTo(data?.Zone));
+        Assert.IsTrue(data?.HistoryData.Count() > 0);
+        var dataPoint = data?.HistoryData.First();
+        Assert.That(new DateTimeOffset(2099, 1, 1, 0, 0, 0, TimeSpan.Zero), Is.EqualTo(dataPoint?.DateTime));
+        Assert.That(new DateTimeOffset(2099, 1, 1, 0, 0, 0, TimeSpan.Zero), Is.EqualTo(dataPoint?.UpdatedAt));
+        Assert.That(new DateTimeOffset(2099, 1, 1, 0, 0, 0, TimeSpan.Zero), Is.EqualTo(dataPoint?.CreatedAt));
+        Assert.That(dataPoint?.CarbonIntensity.ToString(), Is.EqualTo("999"));
+        Assert.That(dataPoint?.EmissionFactorType, Is.EqualTo(EmissionsFactor.Lifecycle));
+        Assert.IsFalse(dataPoint?.IsEstimated);
+        Assert.IsEmpty(dataPoint?.EstimationMethod);
+        Assert.That(new DateTimeOffset(2099, 1, 1, 0, 0, 0, TimeSpan.Zero), Is.EqualTo(dataPoint?.CreatedAt));
+    }
+
+    [TestCase("x-token-header", "faketoken", TestName = "Client config test, token header + token value valid")]
+    [TestCase("", "", TestName = "Client config test, Empty token header + empty token value valid")]
+    public void ClientTest_With_ServiceCollection_VariedConfiguration_DoesNotThrow(string tokenHeader, string tokenValue)
```

Comment:
These `ServiceCollection` tests feel out of place.  Should they be in their own file?

It's not quite clear what the intent of the test are. It feels like they should maybe be direct `ElectricityMapsClientConfiguration_Validate` tests and there should be 1 ServiceCollection test that confirms that `Validate` is called, rather than doing all of this setup.